src/constraints/investment.jl

```diff
@@ -1,62 +1,131 @@
 export add_investment_constraints!
 
 """
-    add_investment_constraints!(graph, Ai, Ase, Fi, assets_investment, assets_investment_energy, flows_investment)
+    add_investment_constraints!(connection, variables)
 
-Adds the investment constraints for all asset types and transport flows to the model
+Adds the investment constraints bounds for all asset types and flows
 """
-function add_investment_constraints!(graph, sets, variables)
-    # TODO: Since this function is defining bound constraints, it doesn't need the `model`
-    # When we refactor the signatures to look the same, we should consider naming it differently
+function add_investment_constraints!(connection, variables)
+    # TODO: When we refactor the signatures to look the same, we should consider naming it differently
     # TODO: Verify if it's possible and reasonable to move the bound definition to when the
-    # indices are created
-    # unpack from sets
-    Ai = sets[:Ai]
-    Ase = sets[:Ase]
-    Fi = sets[:Fi]
-    Y = sets[:Y]
-
-    assets_investment = variables[:assets_investment].lookup
-    assets_investment_energy = variables[:assets_investment_energy].lookup
-    flows_investment = variables[:flows_investment].lookup
 
     # - Maximum (i.e., potential) investment limit for assets
-    for y in Y, a in Ai[y]
-        if graph[a].capacity > 0 && !ismissing(graph[a].investment_limit[y])
-            bound_value = _find_upper_bound(graph, y, a)
-            JuMP.set_upper_bound(assets_investment[y, a], bound_value)
+    constraint_indices = _create_asset_investment_indices(connection)
+    filtered_indices =
+        filter(row -> row.capacity > 0 && !ismissing(row.investment_limit), constraint_indices)
```

Comment:
Instead of return `DataFrame`, we can return the query directly. This makes it more memory efficient. Then, you also don't use `for row in eachrow(table)`, and use directly `for row in table` (or the zip equivalent).

Then, filtering like this doesn't work, and instead we filter inside the loop - I'll also comment something related below